walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```
@@ -14,10 +14,80 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return possibleTimes;
+    }
+
+    possibleTimes.add(TimeRange.WHOLE_DAY);
+
+    if (request.getAttendees().size() == 0) {
+        return possibleTimes;
```

Should you instead throw an error here?

---

I think one of the tests implied that the return value should be an empty collection